internal/xds/clients/internal/buffer/unbounded.go

```diff
@@ -114,3 +115,23 @@ func (b *Unbounded) Close() {
 		close(b.c)
 	}
 }
+
+// Reset clears all buffered data in the unbounded buffer. This does not close
+// the buffer, and new data may be Put() into it after a call to this method.
+//
+// It's expected to be used in scenarios where the buffered data is no longer
+// relevant, and needs to be cleared.
+func (b *Unbounded) Reset() {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	if b.closing {
```

Comment:
would we also need to consider (return early) for the case when `b.closed` is true? 
Although this unbound package and method is exported, but it is internal, so I think there would be little room for misuse.